### ISSUE TITLE
Package marked as non-deterministic due to source file in temp directory

### ISSUE THREAD

**Issue by drewnoakes** [NuGetPackageExplorer/NuGetPackageExplorer#927]
**Type**:

- Question

**NPE version**: NuGet Package Explorer - Store - X64 - (5.6.40+0baf79cb07)

**OS version**: Windows 10 - 1909

**Installed from**: Microsoft Store

I've been trying to work out why NPE says my package is not marked as deterministic. Stepping through the code it appears `AssemblyDebugData.CalculateSourcesDeterministic` is returning `false` as one of the sources has path `%LocalAppData%\Temp\.NETFramework,Version=v4.5.AssemblyAttributes.cs`. This happens for three of my four target frameworks (the odd one out being `net35`).

https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/blob/674ebaf309eadc2a50bb97bcb8fb294e8bbe7ef9/Core/AssemblyMetadata/AssemblyDebugData.cs#L54-L57

The file itself contains:

```c#
// <autogenerated />
using System;
using System.Reflection;
[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5", FrameworkDisplayName = ".NET Framework 4.5")]
```

There are various other files in that folder, for different targets:

- `.NETCoreApp,Version=v1.0.AssemblyAttributes.cs`
- `.NETCoreApp,Version=v1.1.AssemblyAttributes.cs`
- `.NETCoreApp,Version=v2.1.AssemblyAttributes.cs`
- `.NETCoreApp,Version=v2.2.AssemblyAttributes.cs`
- `.NETCoreApp,Version=v3.1.AssemblyAttributes.cs`
- `.NETCoreApp,Version=v5.0.AssemblyAttributes.cs`
- `.NETFramework,Version=v4.0.AssemblyAttributes.cs`
- `.NETFramework,Version=v4.5.1.AssemblyAttributes.cs`
- `.NETFramework,Version=v4.5.AssemblyAttributes.cs`
- `.NETFramework,Version=v4.6.1.AssemblyAttributes.cs`
- `.NETFramework,Version=v4.7.2.AssemblyAttributes.cs`
- `.NETStandard,Version=v1.3.AssemblyAttributes.cs`
- `.NETStandard,Version=v2.0.AssemblyAttributes.cs`

NPE doesn't visualise the non-deterministic file anywhere, which meant having to use the debugger to find out which it was.

Does this mean my build really isn't deterministic, or should the check be improved? NPE explains I should set `Deterministic` and `ContinuousIntegrationBuild`, both of which I have. Is there some other setting I need?

**Comment by clairernovotny** [comment]
Unfortunately there's a bug in the 3.x SDK's that will be fixed in 3.1.300. Until then, there's a workaround here: dotnet/sourcelink#572

I'm open to ways on how to communicate this issue to folks in the interim.

**Comment by drewnoakes** [comment]
The workaround did the trick.

If this is just a point-in-time thing, then it's probably easiest to wait until 3.1.300 is released then bump the recommended SDK version in the tooltip. I see there was one other report of this here (sorry for missing the dupe before). Otherwise I'm not really sure what to do beyond linking to documentation/an issue with further info.
